### DISCUSSION TITLE
Airflow helm chart MEMORY LEAK!!!

### DISCUSSION THREAD

**Discussion by tamis-laan** [apache/airflow#28597] [General]
### Official Helm Chart version

1.7.0 (latest released)

### Apache Airflow version

latest

### Kubernetes Version

Major:"1", Minor:"25", GitVersion:"v1.25.2

### What happened

After installing the airflow chart the machines memory is filled up. 

### How to reproduce

Run the following script for a basic airflow setup:

``` bash
#! /bin/bash

# Add helm charts
helm repo add apache-airflow https://airflow.apache.org

# Create a cluster
kind create cluster --name experiment

# Set the context
kubectl config set-context experiment

# Install airflow operator
helm upgrade --install airflow apache-airflow/airflow --namespace airflow --create-namespace
```

After this runs the memory will fill up.
![screenshot](https://user-images.githubusercontent.com/3178666/209540140-0373a24d-4bdb-473c-b3b7-4ce6fee835df.png)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Clearly SOMETHING fills SOME memory. But what it is, whether this memory is taken by anythong in a chart, whether this is actually a memory that cannot be reclaimed and whether it is at all airflow is not known from what you posted. 

Please provide more details as I could not reproduce it the way you explained it. 

More details needed and more discussion likely if you want us to help you with diagnosing your memory issues.

**Comment by potiuk** [comment]
Converting it into discussion at least until you provide some more details/evidences.

**Comment by potiuk** [comment]
BTW. Usually Yellow color in htop indicates a "cache" memory. And In case you are not aware, cache memory in Linux is perfectly fine to be taking 100% of memory (if you have many files opened and read/written since Linux reboot time, all those files are memory-mapped and take cache). Generally speaking, cache memory will grow continuously (unless you limit it in the kernel setting) but since it is immediately reclaimable when needed, it is not a problem (it is there to speed up file operations).

So you should go deeper and perform more checks on whether your memory is actuallly taken and what kind of memory and which processes are taking it and whether you have a problem at all.

But this investigation is on you.

**Reply by tamis-laan** [reply]
How would I be able to investigate this further? 

Running the helm chart as described nukes my computer within 10 seconds causing it slow down to a crawl and eventually crash...

Also I'm not familiar with the internals of airflow, so no idea where to start.

**Reply by potiuk** [reply]
I think the same way as any other app. I cannot reproduce it by following your steps so if you can tell me how to reproduce it, I might be able to help. But I can't reproduce it, so 🤷

You should try to see which of the processes take the memory and debug it further. You must have something in your system that triggers it - but how - it is hard to say. 

Kind is a development tool and maybe you have some strange setup on your system that triggers it. 

You can also try other kubernetes implementations - there are many. The best way to diagnose such issue is to enable/disable or exchange parts of the system one-by-one. For example you can take a completely different help app and install it and see if it behaves similarly, replace k8s installation etc. and narrow down  the root cause.

That's what I'd in your place at least.

**Reply by potiuk** [reply]
Also try it simply on a completely different machine - even one launched temporarily in the cloud  - to see that it works. And try to compare what is different on your machine and the one that it works on. Again that would be the step I'd do if I had similar problem as you have
